### PR TITLE
simibc: improves package documentation for simibc

### PR DESCRIPTION
**Pull request by danwt**
# Description

Adds a README, doc.go and many more and improved docstrings to testutils/simibc. There are no functional changes or public API changes.

Please note that this is the first of improvements to be made, but another PR (https://github.com/cosmos/interchain-security/pull/658) is using the public API, so I want to wait until that is moved before making any functional changes. 

Also, I want to add an example test showing in a simpl way how things can be used, but I want to tackle some higher priority things before I do that.

## Type of change

If you've checked more than one of the first three boxes, consider splitting this PR into multiple PRs!

- [x] `Refactor`: Changes existing code style, naming, structure, etc.
- [x] `Docs`: Adds documentation

### CODE REVIEW

**Review by smarshall-spitzbart — APPROVED** [review]
Comments all seem very useful, and nothing seems like it's regressing diff test functionality. LGTM!

**Comment by sainoe** [comment]
Let's merge this after #692 since it's a lower priority

File: tests/difference/core/model/src/traceUtil.ts
Comment (smarshall-spitzbart) [reply tests/difference/core/model/src/traceUtil.ts:35]: Once that's decided, we're probably good to merge this one

Comment (MSalopek) [reply tests/difference/core/model/src/traceUtil.ts:35]: That will include the full `git diff` output at the header of the `traces.json`.

It does seem like a nice feature, but I don't see a purpose of including the full diff inside a trace file. Commit hash should be enough and can be used to obtain a diff.

**Review by MSalopek — APPROVED** [review]

**Pull request merged**
